cfme/infrastructure/pxe.py

```diff
@@ -574,6 +574,44 @@ def delete(self, cancel=False, *ct_objs):
             else:
                 navigate_to(ct_obj, 'Details')
 
+    def copy(self, ct_obj, name=None, description=None, cancel=False):
+        """
+         This method is used to copy a Customization Template server via UI.
+
+         Args:
+             ct_obj : This field is the object of Customization Template.
+             name (str): This field contains the name of the newly copied Customization Template.
+             description (str) : This field contains the description of the newly
+              copied Customization Template.
+             cancel (bool): It's used for flag to cancel or not the copy operation.
+        """
+
+        view = navigate_to(ct_obj, 'Details')
+        view.toolbar.configuration.item_select("Copy this Customization Template")
+        view = ct_obj.create_view(PXECustomizationTemplateAddView)
+        if name is not None:
+            view.fill({'name': name})
+        if description is not None:
+            view.fill({'description': description})
+        else:
+            description = 'Copy of {}'.format(ct_obj.description)
+            view.fill({'description': description})
+            name = 'Copy of {}'.format(ct_obj.name)
+
+        customization_templates = self.instantiate(name, description, ct_obj.script_data,
+                                                   ct_obj.image_type, ct_obj.script_type)
+        if cancel:
+            view.cancel.click()
+            msg = 'Add of new Customization Template was cancelled by the user'
+        else:
+            view.add.click()
+            if self.appliance.version < 5.9:
+                msg = 'Customization Template "{}" was saved'.format(name)
+            else:
+                msg = 'Customization Template "{}" was saved'.format(description)
+        view.flash.assert_success_message(msg)
```

Comment:
**REQUIRED**: could you just assert_no_errors() here?
and add separate test that would check flash message after copy

Reply:
@lkhomenk : I think the correct flash message is part of copy method as the copy operation ends with the flash message.

Reply:
correct flash message shouldn't block other testing. separate for flash should assert flash message text like this https://github.com/lkhomenk/integration_tests/blob/d54791a439584da3a912b483e6bb417f4c60f84d/cfme/tests/cloud/test_providers.py#L33.